### PR TITLE
Add 7777

### PR DESCRIPTION
**Pull request by georgeboot**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

After making all changes to a cask, verify:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] There are no [open pull requests](https://github.com/Homebrew/homebrew-cask/pulls) for the same update.
- [x] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#stable-versions) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md).
- [x] `brew cask audit --new-cask {{cask_file}}` worked successfully.
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask).

### CODE REVIEW

File: Casks/7777.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+cask "7777" do
+  version "1.0.3"
+  sha256 "f74b2cb2ef792fc5ab81b5955a02d593dffe1972af61e20fca8a711ee017581b"
+
+  url "https://releases.port7777.com/1.0.3/macos/7777"
```
Comment (miccal) [inline Casks/7777.rb:5]: How did you find this `url` without payment:

Screen Shot 2020-11-19 at 09 58 16

According to their [GitHub support page](https://github.com/whilenull/7777-support#macos) the download link is simply `https://releases.port7777.com/latest/macos/7777`.

Comment (mnapoli) [reply Casks/7777.rb:5]: Hi, 7777 author here. After looking at the documentation of Brew cask I would recommend:

```suggestion
  url "https://releases.port7777.com/#{version}/macos/7777"
```

That will simplify maintenance. I guess installing the `latest` version would work (wouldn't require a PR to update the `.rb` file for new versions), but isn't that contrary to the fact that we specify a specific version above?

> How did you find this url without payment:

The URL is not private, the app requires a license to run. So the process of installing the app via `brew cask` makes sense, you would then run `7777 configure` to setup the license.

File: Casks/7777.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+cask "7777" do
+  version "1.0.3"
+  sha256 "f74b2cb2ef792fc5ab81b5955a02d593dffe1972af61e20fca8a711ee017581b"
```
Comment (mnapoli) [inline Casks/7777.rb:3]: Hi, first time setting up a brew cask!

If I understand correctly, we need to update these 2 fields every time we publish a new version, correct?

Is there any way to automate it? We use GitHub Actions for the release process, we could run anything after uploading the release. Would automatically opening a pull request make sense? That sounds really annoying for you, maintainers. And not automated.

I couldn't find anything about this in the documentation.

File: Casks/7777.rb (hunk 1 above)
Comment (georgeboot) [reply Casks/7777.rb]: @mnapoli update can be triggered semi automated: https://github.com/Homebrew/homebrew-cask/blob/master/CONTRIBUTING.md

As far as I know, there is no 100% automated way, but I believe the cask maintainers are looking at it.

File: Casks/7777.rb (hunk 2 above)
Comment (reitermarkus) [reply Casks/7777.rb:3]: > Would automatically opening a pull request make sense?

That's exactly what you should do. Take a look at the `brew bump-cask-pr` command.

> That sounds really annoying for you, maintainers. And not automated.

As long as only the version and checksum change, our bot will automatically merge it if it passes CI.

Comment (mnapoli) [reply Casks/7777.rb:3]: Thanks for the details.

I'm looking into automating all that. Could we please hold merging this PR until I do? I don't want users to download outdated versions.

Comment (vitorgalvao) [reply Casks/7777.rb:3]: > I'm looking into automating all that.

It’s literally just running `brew bump-cask-pr 7777 --version <NEW_VERSION_HERE>`.

Comment (mnapoli) [reply Casks/7777.rb:3]: Right, but the script has to be authenticated with GitHub right? So we need to create a GitHub bot account too I guess.

And the script needs to run somewhere, so the CI needs to be edited.

**Comment by miccal** [comment]
Thank you @georgeboot.

**Review by miccal — APPROVED** [review]

**Comment by mnapoli** [comment]
> Could we please hold merging this PR until I do?

Welp there goes that.

**Comment by georgeboot** [comment]
> > Could we please hold merging this PR until I do?
> 
> 
> 
> Welp there goes that.

You can easily run that bump version command by hand, until you have it automated, right? It's not like manual downloaders will never run an older version. Not that big of an issue I would say.

**Comment by miccal** [comment]
> > Could we please hold merging this PR until I do?
> 
> Welp there goes that.

Apologies @mnapoli, I missed your comment when I merged this.

**Comment by vitorgalvao** [comment]
> Not that big of an issue I would say.

Agreed. The cask itself was good to go, so no need in keeping the PR indefinitely open waiting for something external.

**Pull request merged**